src/components/CardList/index.jsx

```diff
@@ -1,30 +1,29 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import Movie from '../Card';
-import fetchFrom from '../../services';
+import { fetchMovieByGenre, fetchPopular } from '../../services';
 import './styles.css';
 
 function CardList({
-  id,
   title,
-  url,
   options,
+  setMovieInfo,
 }) {
   const [movieList, setMovieList] = useState([]);
-  const [maxPages, setMaxPages] = useState(0);
-  const [currentPage, setCurrentPage] = useState(options.page);
   const initialPosition = 40;
-  const idName = `${id}-cardList`;
-  const containerName = `${id}-containerCardList`;
+  const containerRef = useRef();
+  const cardListRef = useRef();
   let direction = '';
   let myTimer = null;
-  let isFetching = false;
 
   const fetchMovies = async (fetchOptions) => {
-    isFetching = true;
-    const movies = await fetchFrom(url, fetchOptions);
-    setMovieList([...movieList, ...movies.results]);
-    setMaxPages(movies.total_pages);
-    isFetching = false;
+    let movies;
+    if (options.genre !== null) {
```

Comment:
in this case if there is no genre it will render a popular movies list, we can potentially have more than one popular list of movies, i think the best would be to not render the component if there is no genre and make popular a "genre"